src/black/parsing.py

```diff
@@ -110,6 +110,10 @@ def lib2to3_unparse(node: Node) -> str:
     return code
 
 
+class ASTSafetyError(Exception):
```

Comment:
This should probably inherit from `AssertionError` since you are replacing `raise AssertionError` with this new exception.

Reply:
We don't provide compatibility guarantees for this function. I don't think inheriting from AssertionError makes semantic sense.

Reply:
Thanks for the response. That makes sense considering that many people probably don't use this functionality directly and I maybe agree inheriting from `AssertionError` is odd from a semantic/purity standpoint, even if it's less practical for compatibility (which as you mention is a non-goal here).